endpoints/openrtb2/auction.go

```diff
@@ -434,6 +439,36 @@ func validateSChains(req *openrtb_ext.ExtRequest) error {
 	return err
 }
 
+// validateCustomRates throws a bad input error if any of the 3-digit currency codes found in
+// the bidRequest.ext.prebid.currency field is invalid, malfomed or does not represent any actual
+// currency. No error is thrown if bidRequest.ext.prebid.currency is invalid or empty.
+func validateCustomRates(bidReqCurrencyRates *openrtb_ext.ExtRequestCurrency) error {
+	if bidReqCurrencyRates == nil {
+		return nil
+	}
+
+	customRatesOnly := bidReqCurrencyRates.UsePBSRates != nil && !(*bidReqCurrencyRates.UsePBSRates)
+	if customRatesOnly && len(bidReqCurrencyRates.ConversionRates) == 0 {
+		return &errortypes.BadInput{Message: "Required custom currency rates field is empty"}
```

Comment:
In case `usePbseRates` is set to `false` and the customRates map is empty, PBS-Java uses PBS rates anyways:
```
176     /**
177      * Converts price from bidCurrency to adServerCurrency using rates and usepbsrates flag defined in request.
178      * If usepbsrates is true it takes rates from prebid server, if false from request. Default value of usepbsrates
179      * is true.
180      * Throws {@link PreBidException} in case conversion is not possible.
181      */
182     private BigDecimal convertCurrency(BigDecimal price, Map<String, Map<String, BigDecimal>> requestCurrencyRates,
183                                       String adServerCurrency, String bidCurrency, Boolean usepbsrates) {
184         // use Default USD currency if bidder left this field empty. After, when bidder will implement multi currency
185         // support it will be changed to throwing PrebidException.
186         final String effectiveBidCurrency = bidCurrency != null ? bidCurrency : DEFAULT_BID_CURRENCY;
187 +--  7 lines: if (Objects.equals(adServerCurrency, effectiveBidCurrency)) {-------------------------------------------------------------------------------------------------------------------------------------------------------------
194
195         if (BooleanUtils.isFalse(usepbsrates)) {
196             firstPriorityRates = requestCurrencyRates;   // <-- what we call custom currency rates
197             secondPriorityRates = externalCurrencyRates; // <-- PBS rates
198         } else {
199             firstPriorityRates = externalCurrencyRates;
200             secondPriorityRates = requestCurrencyRates;
201         }
202
203         final BigDecimal conversionRate = getConversionRateByPriority(firstPriorityRates, secondPriorityRates,
204                 adServerCurrency, effectiveBidCurrency);
205
206         if (conversionRate == null) {
207             throw new PreBidException(
208                     String.format("Unable to convert from currency %s to desired ad server currency %s",
209                             effectiveBidCurrency, adServerCurrency));
210         }
211
212         return price.divide(conversionRate, DEFAULT_PRICE_PRECISION, RoundingMode.HALF_EVEN);
213     }
src/main/java/org/prebid/server/currency/CurrencyConversionService.java
```

Reply:
I'll go ahead and add the logic for disabling rate conversion when `usePbsRates` is `false` and the custom currency rate map is empty.

Reply:
Done. Added a couple of test cases when conversions were disabled